dot/parachain/provisioner/prioritized_disputes_selection_test.go

```diff
@@ -384,6 +537,625 @@ func TestPartitioningDuplicatedDispute(t *testing.T) {
+	sessionIdx := parachaintypes.SessionIndex(0)
+	lf := newLeaf()
+	dummyReceiptv2 := dummyCandidateReceiptV2(lf.Hash)
+
+	for i := 0; i < disputeCount; i++ {
+		rndHash := make([]byte, 32)
+		_, err := rand.Read(rndHash)
+		require.NoError(t, err)
+
+		candidate := parachaintypes.CandidateHash{Value: common.Hash(rndHash)}
+		disputeStatus := *setEnumVariant[*parachaintypes.DisputeStatus](parachaintypes.Active{})
+		td.addOffchainDispute(
+			t,
+			sessionIdx,
+			candidate,
+			disputeStatus,
+			localVotesCount,
+			fromCandidateRecepitV2ToV1(dummyReceiptv2),
+		)
+		td.addOnchainDispute(t, sessionIdx, candidate, disputeStatus, onchainVotesCount)
+	}
+	diff := (localVotesCount - onchainVotesCount) * disputeCount
+	return sessionIdx, diff
+}
+
+// addUnconfirmedDisputesUnconcludedOnchain adds unconfirmed disputes that are unconcluded onchain.
+// Returns (sessionIdx, (localVotesCount - onchainVotesCount) * disputeCount).
+func (td *TestDisputes) addUnconfirmedDisputesUnconcludedOnchain(
+	t *testing.T,
+	disputeCount int,
+) (parachaintypes.SessionIndex, int) {
+	localVotesCount := td.ValidatorsCount * 90 / 100
+	onchainVotesCount := td.ValidatorsCount * 40 / 100
+
+	fmt.Println("onchain votes count", onchainVotesCount)
+
+	sessionIdx := parachaintypes.SessionIndex(1)
+	lf := newLeaf()
+	dummyReceipt := dummyCandidateReceiptV2(lf.Hash)
+	for i := 0; i < disputeCount; i++ {
+		rndHash := make([]byte, 32)
+		_, err := rand.Read(rndHash)
+		require.NoError(t, err)
+		candidate := parachaintypes.CandidateHash{Value: common.Hash(rndHash)}
+		disputeStatus := *setEnumVariant[*parachaintypes.DisputeStatus](parachaintypes.Active{})
+		td.addOffchainDispute(
+			t,
+			sessionIdx,
+			candidate,
+			disputeStatus,
+			localVotesCount,
+			fromCandidateRecepitV2ToV1(dummyReceipt),
+		)
+		td.addOnchainDispute(t, sessionIdx, candidate, disputeStatus, onchainVotesCount)
+	}
+	diff := (localVotesCount - onchainVotesCount) * disputeCount
+	return sessionIdx, diff
+}
+
+func (td *TestDisputes) addConfirmedDisputesUnkonwOnChain(
+	t *testing.T,
+	disputeCount int,
+) (parachaintypes.SessionIndex, int) {
+	localVotesCount := td.ValidatorsCount * 90 / 100
+	sessionIdx := parachaintypes.SessionIndex(2)
+	lf := newLeaf()
+	dummyReceiptv2 := dummyCandidateReceiptV2(lf.Hash)
+
+	for i := 0; i < disputeCount; i++ {
+		rndHash := make([]byte, 32)
+		_, err := rand.Read(rndHash)
+		require.NoError(t, err)
+
+		candidate := parachaintypes.CandidateHash{Value: common.Hash(rndHash)}
+		disputeStatus := *setEnumVariant[*parachaintypes.DisputeStatus](parachaintypes.Confirmed{})
+		td.addOffchainDispute(
+			t,
+			sessionIdx,
+			candidate,
+			disputeStatus,
+			localVotesCount,
+			fromCandidateRecepitV2ToV1(dummyReceiptv2),
+		)
+	}
+	return sessionIdx, localVotesCount * disputeCount
+}
+
+func (td *TestDisputes) addConcludedDisputesKnownOnchain(
+	t *testing.T,
+	disputeCount int,
+) (parachaintypes.SessionIndex, int) { //nolint:unparam
```

Comment:
Does this linter flag have any effect on return values?

Reply:
yup, the linter identifies that the returned valued was never being used in any of the calls